src/sql/workbench/services/connection/browser/connectionWidget.ts

```diff
@@ -1209,26 +1215,26 @@ export class ConnectionWidget extends lifecycle.Disposable {
 				model.authenticationType = this.authenticationType;
 				model.azureAccount = this.authToken;
 				model.savePassword = this._rememberPasswordCheckBox.checked;
-				model.connectionName = this.connectionName;
 				model.databaseName = this.databaseName;
-				if (this._customOptionWidgets) {
-					this._customOptionWidgets.forEach((widget, i) => {
-						model.options[this._customOptions[i].name] = widget.value;
-					});
-				}
-				if (this._serverGroupSelectBox) {
-					if (this._serverGroupSelectBox.value === this.DefaultServerGroup.name) {
-						model.groupFullName = '';
-						model.saveProfile = true;
-						model.groupId = this.findGroupId(model.groupFullName);
-					} else if (this._serverGroupSelectBox.value === this.NoneServerGroup.name) {
-						model.groupFullName = '';
-						model.saveProfile = false;
-					} else if (this._serverGroupSelectBox.value !== this._addNewServerGroup.name) {
-						model.groupFullName = this._serverGroupSelectBox.value;
-						model.saveProfile = true;
-						model.groupId = this.findGroupId(model.groupFullName);
-					}
+			}
+			model.connectionName = this.connectionName;
+			if (this._customOptionWidgets) {
```

Comment:
Custom option widgets are not rendered when using connection string, fyi.. You can move this block back to the else case.

Reply:
Right, not sure why I missed that, I just assumed that custom options would always be visible.